### ISSUE TITLE
Session Management

### ISSUE THREAD

**Issue by ManiDeepakReddyAila** [BigDataIA-Spring2023-Team-05/Assignment-03#30]
- See a lot of issues related to Session Management, all of these could be avoided by managing it right.

1. Once user logout, login form is not fetched until we do another call.
2. Users are expected to reload the application to be able to register a new user again.
3. Was able to access forgot password page even when user's session is active and still logged in.
4. Got this error when trying to access NEXRAD page without logging in 
    `HTTPError: HTTP Error 503: Service Unavailable`

**Comment by ManiDeepakReddyAila** [comment]
<img width="1440" alt="Screen Shot 2023-03-17 at 12 21 42 PM" src="https://user-images.githubusercontent.com/114799981/225973638-2c3447ea-8e55-455b-8bb1-c6a644eb2c99.png">

**Comment by Charan-Patnaik** [comment]
1.  Yes , this would be additional functionality , to take into account . Though was not added since this does not interfere with project requirements.
2. Yes, this is was the product design choice to stream line the user journey , given already preexisting users who is logged in would not try to register as user .
3. Yes , this also design choice that we made , where a user post login can change his/her password . Will add the sub-header title as 'Forgot password / Update password'
4. We could not replicate this issue , post the steps to replicate the issue.
Here is the video which usages without issues.
https://northeastern.sharepoint.com/:v:/s/BigData80/ER1Aw_SdVs1Drg0kgP8fNpQBGBaz1yOleXXqmfcsznW75Q?e=flGKUW

**Comment by ManiDeepakReddyAila** [comment]
> Yes , this would be additional functionality , to take into account . Though was not added since this does not interfere with project requirements.
Yes, this is was the product design choice to stream line the user journey , given already preexisting users who is logged in would not try to register as user .
Yes , this also design choice that we made , where a user post login can change his/her password . Will add the sub-header title as 'Forgot password / Update password'
We could not replicate this issue , post the steps to replicate the issue.
Here is the video which usages without issues.
[https://northeastern.sharepoint.com/✌️/s/BigData80/ER1Aw_SdVs1Drg0kgP8fNpQBGBaz1yOleXXqmfcsznW75Q?e=flGKUW](https://northeastern.sharepoint.com/:v:/s/BigData80/ER1Aw_SdVs1Drg0kgP8fNpQBGBaz1yOleXXqmfcsznW75Q?e=flGKUW)

I won't agree with the above points

1. User once logged out should be able to register as a new user, he shouldn't be forced to reload the application to get the register form. ( Project requirements also talk about multiple users, not one user per application launch )
2. Same is the case for login, if I logout by mistake or by choice I should be given option to login again ( Again it is not about Project Requirements )
3. I'm not sure if we should allows user to change his password when there is an active session. What would happen to the active session if user updates the password in between?

**Comment by Charan-Patnaik** [comment]
1.  Yes , for registration part i agree with you partly  , the above mentioned point earlier on registration was made under the assumption that user is logged in that means there is not requirement for the user to re-register again. 
2.  With regards to login page ,  your comments are noted.
3. With regards to the forgot password .. coming back to the root cause :  'Was able to access forgot password page even when user's session is active and still logged in' .. we could not replicate it , below is the video demonstrating the same.
Note :  Earlier on forgot password ,  sent out the comment with different assumption than it was stated by you @ManiDeepakReddyAila  .. ignore that.
https://northeastern.sharepoint.com/:v:/s/BigData80/EXStvl4jubxEqZzCfbw7uXoBkMofOZrrKqNR15RrLDijAg?e=VKCdwq
